### ISSUE TITLE
Error handling in the spawned Benchmark process

### ISSUE THREAD

**Issue by mattwarren** [PerfDotNet/BenchmarkDotNet#170]
At the moment our BenchmarkProgram.txt template looks like so:

``` csharp
public static void Main(string[] args)
{
	try
	{
		...

		instance.RunBenchmark();
	}
	catch (Exception ex)
	{
		System.Console.WriteLine(ex);
		throw;
	}
}
```

The problem is that when we run a benchmark that throws an exception, the following happens:

![image](https://cloud.githubusercontent.com/assets/157298/15111799/7baeb03c-15e1-11e6-8174-db5ff1c7be22.png)

I.e. you get the "Do you want to send ..." modal dialogue box. This is also a problem when running our integration tests, since I added [BaselineDeltaResultExtenderHandlesBenchmarkErrorTest](https://github.com/PerfDotNet/BenchmarkDotNet/commit/3576d4f4ccd619bcdff79742abb4d9496da7b0fc)

### Proposed solution

Ensure that exceptions don't escape from the generate benchmark, but instead we return an error code, i.e.

``` csharp
public static int Main(string[] args)
{
	try
	{
		...

		instance.RunBenchmark();
		return 0;
	}
	catch (Exception ex)
	{
		System.Console.WriteLine(ex);
		return -1;
	}
}
```

Note that we will still print the exception in the logger output:

![image](https://cloud.githubusercontent.com/assets/157298/15112917/1e8af62e-15e9-11e6-817c-4c31d6cb4096.png)

and the validators/analyser still picks up that the benchmark has a problem:

![image](https://cloud.githubusercontent.com/assets/157298/15111837/bc6d5fec-15e1-11e6-81cd-8d1fd9985a02.png)

@AndreyAkinshin @adamsitnik is this a good idea, will it break anything?

**Comment by AndreyAkinshin** [comment]
LGTM.

**Comment by adamsitnik** [comment]
Good idea! :+1:

**Comment by mattwarren** [comment]
Okay, I'll implement it. But I won't check it in until after you've published 0.9.6 to NuGet. 

I don't want to cause any more churn, in case there's a problem.
